frappe/utils/data.py

```diff
@@ -635,76 +637,110 @@ def format_datetime(datetime_string: DateTimeLikeObject, format_string: str | No
+	for unit, value in duration.items():
+		if value > 0:
+			abbr = DURATION_DATA[unit]["abbr"]
+			context = unit.upper() + " (Field: Duration)"
+			components.append(str(value) + frappe._(abbr, context=context))
+	return " ".join(components)
 
-	if hide_days:
-		total_duration["hours"] = math.floor(seconds / 3600)
-		total_duration["days"] = 0
-
-	duration = ""
-	if total_duration:
-		if total_duration.get("days"):
-			duration += str(total_duration.get("days")) + "d"
-		if total_duration.get("hours"):
-			duration += " " if len(duration) else ""
-			duration += str(total_duration.get("hours")) + "h"
-		if total_duration.get("minutes"):
-			duration += " " if len(duration) else ""
-			duration += str(total_duration.get("minutes")) + "m"
-		if total_duration.get("seconds"):
-			duration += " " if len(duration) else ""
-			duration += str(total_duration.get("seconds")) + "s"
 
-	return duration
+def seconds_to_duration(
+	seconds: int, duration_options: dict[str, bool] | None = None
+) -> dict[str, int]:
+	"""
+	Convert the given seconds into a duration object with all unit parts that are not hidden.
+	"""
 
+	if duration_options and not isinstance(duration_options, dict):
+		raise TypeError(
+			f"duration_options should be of type 'dict', not '{type(duration_options).__name__}'"
+		)
 
-def duration_to_seconds(duration):
-	"""Converts the given duration formatted value to duration value in seconds
+	duration = {}
+	remaining_seconds = seconds
+	for unit, data in DURATION_DATA.items():
+		if not duration_options or not duration_options.get(f"hide_{unit}"):
+			duration[unit] = int(remaining_seconds // data["factor"])
+			remaining_seconds %= data["factor"]
+	return duration
 
-	example: converts '3h 34m 45s' to 12885 (value in seconds)
-	"""
-	validate_duration_format(duration)
-	value = 0
-	if "d" in duration:
-		val = duration.split("d")
-		days = val[0]
-		value += cint(days) * 24 * 60 * 60
-		duration = val[1]
-	if "h" in duration:
-		val = duration.split("h")
-		hours = val[0]
-		value += cint(hours) * 60 * 60
-		duration = val[1]
-	if "m" in duration:
-		val = duration.split("m")
-		mins = val[0]
-		value += cint(mins) * 60
-		duration = val[1]
-	if "s" in duration:
-		val = duration.split("s")
-		secs = val[0]
-		value += cint(secs)
 
-	return value
+def duration_to_seconds(duration_str: str) -> int:
+	"""
+	Converts formatted duration string to duration in seconds.
 
+	Args:
+	        str: String with the duration broken down into all unit parts not hidden.
+	Returns:
+	        int: The duration in seconds.
+	"""
 
-def validate_duration_format(duration):
+	total_seconds = 0
+	for unit, data in DURATION_DATA.items():
+		if unit == "minutes":
+			# BC: Understand both "min" and "m", but not "mo".
+			pattern = r"(\d+)m(?!o)"
+		else:
+			pattern = r"(\d+)" + re.escape(data["abbr"])
+		match = re.search(pattern, duration_str)
+		if match:
+			value = int(match.group(1))
+			total_seconds += value * data["factor"]
```

Comment:
Using the conversion table.
Lost the `validate_duration_format()` check here, will add after your review. Don’t want to mess around with the PR in the meantime.